File: front/src/types/Post.ts
import type { Image } from '@/types/Image'
import type { Video } from '@/types/Video'

/**
 * Post type
 */
export interface Post {
  id: string
  date: Date
  title: string
  content: string
  image?: Image
  video?: Video
  created: Date
  updated: Date
}

File: back/myband-profile/src/main/java/net/franz/myband/profile/domain/Post.java
package net.franz.myband.profile.domain;

import lombok.Data;
import net.franz.myband.profile.domain.media.Image;
import net.franz.myband.profile.domain.media.Video;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import java.util.List;

@Data
@Document(collection = "posts")
public class Post extends CollectionItem {

    @Id
    private String id;

    private String profileId;

    private String title;
    private String content;

    private Image image;
    private Video video;

    List<Comment> comments;

}
